src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat.pop());
+  const dateObj = {};
+
+  fromFormat.forEach(function (part, i) {
+    dateObj[part] = dateParts[i];
+  });
+
+  const convertYear = (year, format) => {
+    if (format === 'YYYY') {
+      return year.length === 2 ? (year < 30 ? '20' + year : '19' + year) : year;
+    }
+
+    return year.length === 4 ? year.slice(2) : year;
+  };
+
+  const resultParts = toFormat.slice(0, -1).map((part) => {
```

Comment:
Using `slice(0, -1)` on `toFormat` to exclude the separator assumes that the separator is always the last element. This is not explicitly guaranteed in the task description. It would be safer to filter out non-date parts explicitly based on an understanding of what constitutes a date part (e.g., 'YYYY', 'MM', 'DD').